java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -7576,4 +7581,479 @@ public void geosearchstore(BaseClient client) {
+        for (int i = 0; i < 5; i++) {
+            charMap.put(charMembers[i], String.valueOf(i));
+        }
+
+        // Empty set
+        Object[] result = client.hscan(key1, initialCursor).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Negative cursor
+        result = client.hscan(key1, "-1").get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {}, result[resultCollectionIndex]);
+
+        // Result contains the whole set
+        assertEquals(charMembers.length, client.hset(key1, charMap).get());
+        result = client.hscan(key1, initialCursor).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertEquals(
+                charMap.size() * 2,
+                ((Object[]) result[resultCollectionIndex])
+                        .length); // Length includes the score which is twice the map size
+        final Object[] resultArray = (Object[]) result[resultCollectionIndex];
+
+        final Set<Object> resultKeys = new HashSet<>();
+        final Set<Object> resultValues = new HashSet<>();
+        for (int i = 0; i < resultArray.length; i += 2) {
+            resultKeys.add(resultArray[i]);
+            resultValues.add(resultArray[i + 1]);
+        }
+        assertTrue(
+                resultKeys.containsAll(charMap.keySet()),
+                String.format("resultKeys: {%s} charMap.keySet(): {%s}", resultKeys, charMap.keySet()));
+
+        assertTrue(
+                resultValues.containsAll(charMap.values()),
+                String.format("resultValues: {%s} charMap.values(): {%s}", resultValues, charMap.values()));
+
+        result =
+                client.hscan(key1, initialCursor, HScanOptions.builder().matchPattern("a").build()).get();
+        assertEquals(initialCursor, result[resultCursorIndex]);
+        assertDeepEquals(new String[] {"a", "0"}, result[resultCollectionIndex]);
+
+        // Result contains a subset of the key
+        final HashMap<String, String> combinedMap = new HashMap<>(numberMap);
+        combinedMap.putAll(charMap);
+        assertEquals(numberMap.size(), client.hset(key1, combinedMap).get());
+        String resultCursor = "0";
+        final Set<Object> secondResultAllKeys = new HashSet<>();
+        final Set<Object> secondResultAllValues = new HashSet<>();
+        do {
+            result = client.hscan(key1, resultCursor).get();
+            resultCursor = result[resultCursorIndex].toString();
+            Object[] resultEntry = (Object[]) result[resultCollectionIndex];
+            for (int i = 0; i < resultEntry.length; i += 2) {
+                secondResultAllKeys.add(resultEntry[i]);
+                secondResultAllValues.add(resultEntry[i + 1]);
+            }
+
+            if (resultCursor.equals("0")) {
+                break;
+            }
+
```

Comment:
Why do you need this? The loop will break on that condition anyway on line 7999

Reply:
If you don't check this after the hscan at 7972, and it returned 0, you'd call hscan again at 7985 and start a new cursor. So not strictly necessary but avoids some extra overhead.